tests/foreman/api/test_activationkey.py

```diff
@@ -173,6 +212,24 @@ def test_negative_update_1(self):
                 actual = {attr: getattr(act_key, attr) for attr in want.keys()}
                 self.assertEqual(want, actual)
 
+    def test_negative_update_2(self):
+        """@Test: Create activation key then update its name to an invalid name.
+
+        @Assert: Activation key is created, and its name is not updated.
+
+        @Feature: ActivationKey
+
+        """
+        new_key = entities.ActivationKey().create()
+        for new_name in invalid_names_list():
+            with self.subTest(new_name):
+                with self.assertRaises(HTTPError):
```

Comment:
This HTTP error is expected for which call: update or read? It is better to have just the intended call under the with statement.